### PR TITLE
Add log tag to exchangestore

### PR DESCRIPTION
**Pull request by boaks**
Signed-off-by: Achim Kraus achim.kraus@bosch-si.com

### CODE REVIEW

File: californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java [hunk 1]
```diff
@@ -1330,6 +1342,17 @@ public Builder setCoapStackFactory(CoapStackFactory coapStackFactory) {
...
 		}
 
+		/**
+		 * Set logging tag.
+		 * 
+		 * @param tag logging tag. Defautls to connector's scheme.
+		 * @return this
+		 */
+		public Builder setLoggingTag(String tag) {
+			this.tag = tag;
```
Comment (sbernard31) [inline californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java]: I'm not sure but should we normalize tag here ?

File: californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java [hunk 2]
```diff
@@ -126,6 +133,8 @@ public InMemoryMessageExchangeStore(final NetworkConfig config, TokenGenerator t
 		this.tokenGenerator = tokenProvider;
 		this.endpointIdentityResolver = endpointResolver;
 		this.config = config;
+		this.tag = StringUtil.normalizeLoggingTag(tag);
+		LOGGER.debug("{}using default TokenProvider {}", tag, RandomTokenGenerator.class.getName());
```
Comment (sbernard31) [inline californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java]: default token  provider is maybe not used here.

File: californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java (hunk 1 above)
Comment (boaks) [reply californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java]: Without there, `null`could be applied to use the default tag.

File: californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java (hunk 2 above)
Comment (boaks) [reply californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java]: Ok.

**Pull request merged**